code/modules/client/client_procs.dm

```diff
@@ -453,6 +453,10 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 //////////////
 
 /client/Del()
+	GLOB.clients -= src
+	GLOB.directory -= ckey
+	if(!gc_destroyed)
+		Destroy() //Clean up signals and timers.
```

Comment:
:thonk:

If we're going to call Destroy from Del should the bulk of the code not be moved there?

Reply:
Probably, yeah. I'll do it in a moment.